ginivision/src/test/java/net/gini/android/vision/requirements/CameraResolutionRequirementTest.java

```diff
@@ -12,49 +12,44 @@
 import org.junit.runners.JUnit4;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 @RunWith(JUnit4.class)
 public class CameraResolutionRequirementTest {
 
     @Test
-    public void should_reportUnfulfilled_ifNoPreviewSize_with4to3AspectRatio() {
-        CameraHolder cameraHolder = getCameraHolder(Arrays.asList(createSize(300, 200), createSize(600, 400)), null);
-
-        CameraResolutionRequirement requirement = new CameraResolutionRequirement(cameraHolder);
-
-        assertThat(requirement.check().isFulfilled()).isFalse();
-    }
-
-    @Test
-    public void should_reportUnfulfilled_ifNoPictureSize_with4to3AspectRatio() {
-        CameraHolder cameraHolder = getCameraHolder(null, Arrays.asList(createSize(4000, 2000), createSize(3000, 3000)));
+    public void should_reportUnfulfilled_ifNoPreviewSize_withSameAspectRatio_asLargestPictureSize() {
+        CameraHolder cameraHolder = getCameraHolder(Collections.singletonList(createSize(300, 200)), null);
 
         CameraResolutionRequirement requirement = new CameraResolutionRequirement(cameraHolder);
 
         assertThat(requirement.check().isFulfilled()).isFalse();
+        assertThat(requirement.check().getDetails()).isEqualTo("Camera has no preview resolutions matching the picture resolution 3840x2160");
     }
 
     @Test
     public void should_reportUnfulfilled_ifPictureSize_isSmallerThan8MP() {
         CameraHolder cameraHolder = getCameraHolder(null,
                 Arrays.asList(
                         createSize(400, 300),
-                        createSize(3200, 2400)) //7,68MP
+                        createSize(3200, 2048)) //6,55MP
         );
 
         CameraResolutionRequirement requirement = new CameraResolutionRequirement(cameraHolder);
 
         assertThat(requirement.check().isFulfilled()).isFalse();
+        assertThat(requirement.check().getDetails()).isEqualTo("Largest camera picture resolution is lower than 8MP");
     }
 
     @Test
-    public void should_reportFulfilled_ifPreviewSize_andPictureSize_with4to3AspectRatio_andPictureSize_isLargerThan8MP() {
+    public void should_reportFulfilled_ifPreviewSize_andPictureSize_isLargerThan8MP() {
```

Comment:
Personally, I don't like the naming convention for test names. I know it's in your code style guide, though, so it's ok.
[Android's guide](http://source.android.com/source/code-style.html#javatests-style-rules) uses kind of the same thing with the underscores. However, they start names with "test".
_No action for this, just expression an opinion_